### PR TITLE
More preloading + new website complex

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [ ] This PR addresses one or more open issues that were assigned to me:
      - Ongoing effort to smooth out the feed
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
This PR adds preloading for user and community avatars. It also revamps the `WebsiteIconComplex` to be less line-y and more Apple (and *way* nicer on the backend)

## Screenshots and Videos
Screenshot 2023-07-07 at 12 59 48 PM

### CODE REVIEW

File: Mlem/Models/Trackers/Post Tracker.swift [hunk 1]
```diff
@@ -67,28 +68,50 @@ class PostTracker: FeedTracker<APIPostView> {
...
     private func preloadImages(_ newPosts: [APIPostView]) {
         URLSession.shared.configuration.urlCache = AppConstants.urlCache
-        for post in newPosts {
-            if let thumbnailUrl = post.post.thumbnailUrl {
-                Task(priority: .background) {
-                    await preloadSingleImage(url: thumbnailUrl)
+        for postView in newPosts {
+            // preload user and community avatars--fetching both because we don't know which we'll need, but these are super tiny
+            // so it's probably not an API crime, right?
+            Task(priority: .userInitiated) {
```
Comment (tht7) [inline Mlem/Models/Trackers/Post Tracker.swift:74]: Why user initiated? I think this should be background

**Review by tht7 — APPROVED** [review]

File: Mlem/Models/Trackers/Post Tracker.swift (hunk 1 above)
Comment (EricBAndrews) [reply Mlem/Models/Trackers/Post Tracker.swift:74]: Muscle memory lol

**Comment by tht7** [comment]
We should really swap that to using groupTasks but that can be done later

File: Mlem/Views/Shared/Website Icon Complex.swift [hunk 2]
```diff
@@ -49,79 +49,50 @@ struct WebsiteIconComplex: View {
...
-                    }
-                }
-
-                HStack(alignment: .center, spacing: 0) {
-                    if shouldShowWebsiteFaviconAtAll {
+        VStack(spacing: 0) {
+            if shouldShowWebsitePreviews, let thumbnailURL = post.thumbnailUrl {
+                CachedImageWithNsfwFilter(isNsfw: post.nsfw, url: thumbnailURL)
+                    .frame(maxHeight: 400)
+                    // .cornerRadius(AppConstants.largeItemCornerRadius)
```
Comment (mormaer) [inline Mlem/Views/Shared/Website Icon Complex.swift:56]: 🧹

**Review by mormaer — COMMENTED** [review]
Couple of small questions, nothing major

There are rate limits on the servers which are returned in the `/site` call which we should look to respect soon, the whole pre-loading stuff should likely be moved out of this tracker and into it's own _thing_, but that's a different task...

**Comment by EricBAndrews** [comment]
Converted to draft pending further tidying

**Comment by WestonHanners** [comment]
Do we need to display the license for Nuke in the about screen?

**Review by WestonHanners — APPROVED** [review]

**Pull request merged**